src/interfaces/wasi/wit_generation.rs

```diff
@@ -0,0 +1,269 @@
+// Since wit_generation.rs code is called from the Cargo feature `wasi_impl`,
+// rustc cannot see that code is actually not a dead code
+#![allow(dead_code)]
```

Comment:
so why this file is not under conditional compilation like `wit_implementation.rs` is?

Reply:
well, its implementation is shared with `wit_implementation`. It is a bit tricky since I wanted to factorize all ABE-functions in 1 file : 
- `wit_interface` actually has all the ABE-functions and is used as well for WIT generation
- `wit_implementation` only transfer input parameters to `wit_interface` but also implements the Rust ABE-Bindings